### PR TITLE
New England Great Revival: Mapgen and Missions

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
None

#### Purpose of change
I'm dividing https://github.com/CleverRaven/Cataclysm-DDA/pull/62061 into two parts for easier review, this one only has the mapgen and missions.

#### Describe the solution
This only included the missions and mapgen for God's Chosen. I'm keeping this as a draft pull request for now, so everything can be merged in one go. If or when that happens, I'll close this. The checks for https://github.com/CleverRaven/Cataclysm-DDA/pull/62061 have already completed as green; I'd recommend ignoring the checks on this.

#### Describe alternatives you've considered
Making this a full, seperate PR

#### Testing
None

#### Additional context
I'll link these in #62061

### CODE REVIEW

File: data/json/mapgen/godco/godco_eocs.json [hunk 1]
```diff
@@ -0,0 +1,106 @@
+[
+  {
+    "type": "effect_on_condition",
+    "id": "godco_place_coop",
+    "recurrence": 100,
+    "global": true,
+    "condition": {
+      "and": [
+        { "u_has_var": "darryl_building_coop", "type": "general", "context": "trade", "value": "yes" },
```
Comment (andrei8l) [inline data/json/mapgen/godco/godco_eocs.json:9]: These variables should be stored on the NPC or globally. That way the constructions can still progress after you die and switch to an NPC follower.

Comment (MNG-cataclysm) [reply data/json/mapgen/godco/godco_eocs.json:9]: How do I store the variables globally?

Comment (andrei8l) [reply data/json/mapgen/godco/godco_eocs.json:9]: `{ "global_val": "var", "var_name": "my_var" }`. See an example [here](https://github.com/CleverRaven/Cataclysm-DDA/blob/bd17c48f95b8d7e467a8c413b97ff5299fae0524/data/json/npcs/factions.json#L167).

Comment (MNG-cataclysm) [reply data/json/mapgen/godco/godco_eocs.json:9]: Can timers also be stored globally?

Comment (MNG-cataclysm) [reply data/json/mapgen/godco/godco_eocs.json]: I've changed them to be stored both globally and on the NPCs.

**Review by Maleclypse — COMMENTED** [review]
I'm not having any luck finding the missions to review?

**Comment by MNG-cataclysm** [comment]
> I'm not having any luck finding the missions to review?  

Yeah, I accidentally deleted the missions file. The original pull request still has it, though.

**Comment by Maleclypse** [comment]
> > I'm not having any luck finding the missions to review?
> 
> Yeah, sorry about that; I accidentally deleted the missions file. The original pull request still has it, though.

Mapgen looks good.  I'll review the missions on the original PR and if no changes needed will approve :)

**Comment by MNG-cataclysm** [comment]
Closing as the main PR was merged.

**Pull request closed without merging**